lib/premailer/rails/css_loaders/asset_pipeline_loader.rb

```diff
@@ -5,18 +5,8 @@ module AssetPipelineLoader
         extend self
 
         def load(url)
-          if asset_pipeline_present?
-            file = file_name(url)
-            asset = ::Rails.application.assets.find_asset(file)
-            asset.to_s if asset
-          end
-        end
-
-        def asset_pipeline_present?
-          defined?(::Rails) &&
-            ::Rails.respond_to?(:application) &&
-            ::Rails.application.respond_to?(:assets) &&
-            ::Rails.application.assets
+          file = file_name(url)
+          ::Rails.application.assets_manifest.find_sources(file).first
```

Comment:
@kirs thanks for the PR, looks cool! I only afraid it would break certain kinds of rails apps deployments, e.g. when you have assets on frontend servers only, while upstream servers only have sprockets-manifest present to set correct asset urls (assuming config.assets.digest set to true), in this case `assets_manifest.find_sources(file).first` would throw `Errno::ENOENT: No such file or directory @ rb_sysopen`

Reply:
Should we rescue it then?

I'm assuming that if the server only has manifest, then the former code (`::Rails.application.assets.find_asset(file)`) would also not work.

Am I mistaken?

Reply:
> Should we rescue it then?

yeap, rescue would do the trick

> I'm assuming that if the server only has manifest, then the former code (::Rails.application.assets.find_asset(file)) would also not work.

you're correct, but the thing is asset_pipeline_present? returns false in this case, because ::Rails.application.assets is nil

Reply:
In production `::Rails.application.assets` should be always `nil`, but the manifest can be present with assets or without assets. We can't detect that based on `::Rails.application.assets`, or any other config. The only way is for rescue that (or a more specific exception if we change sprockets)